### PR TITLE
Problem: (fix #305) pystarport missing optional pubkey in config

### LINKED ISSUE #305: Problem: pystarport missing optional pubkey in config
When using pystarport to generate new genesis, one can specify consensus_key `pub` and `priv` in config.
It would be great to add pubkey as well. As some may use tmkms for initial validator, they are not using `priv_validator_key.json`

### PR DESCRIPTION
**Pull request by allthatjazzleo**
Solution: add option for pubkey in pystarport

### CODE REVIEW

File: pystarport/pystarport/cosmoscli.py [hunk 1]
```diff
@@ -159,6 +162,7 @@ def gentx(self, name, coins, min_self_delegation=1):
             home=self.data_dir,
             chain_id=self.chain_id,
             keyring_backend="test",
+            **{k: v for k, v in options.items() if v is not None},
```
Comment (yihuang) [inline pystarport/pystarport/cosmoscli.py:165]:
```suggestion
            pubkey=pubkey,
```

It already filter out `None` values.

**Review by yihuang — APPROVED** [review]
LGTM, small suggestions.

File: pystarport/pystarport/cosmoscli.py [hunk 2]
```diff
@@ -150,7 +150,10 @@ def add_genesis_account(self, addr, coins, **kwargs):
             **kwargs,
         )
 
-    def gentx(self, name, coins, min_self_delegation=1):
+    def gentx(self, name, coins, min_self_delegation=1, pubkey=None):
+        options = dict(
```
Comment (tomtau) [inline pystarport/pystarport/cosmoscli.py:154]: not needed?

**Review by tomtau — CHANGES_REQUESTED** [review]
https://github.com/crypto-com/chain-main/pull/357/checks?check_run_id=1716264450#step:5:49

File: pystarport/pystarport/cosmoscli.py (hunk 2 above)
Comment (allthatjazzleo) [reply pystarport/pystarport/cosmoscli.py]: fixed

**Review by tomtau — APPROVED** [review]

**Pull request merged**